client/qiskit_serverless/utils/runtime_service_client.py

```diff
@@ -0,0 +1,120 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+======================================================================
+Json utilities (:mod:`qiskit_serverless.utils.runtime_service_client`)
+======================================================================
+
+.. currentmodule:: qiskit_serverless.utils.runtime_service_client
+
+Qiskit Serverless runtime client wrapper
+========================================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    ServerlessRuntimeService
+"""
+import os
+import logging
+from typing import Callable, Dict, Sequence, Type, Union, Optional
+
+import requests
+from qiskit_ibm_runtime import QiskitRuntimeService
+from qiskit_ibm_runtime.runtime_job import RuntimeJob
+from qiskit_ibm_runtime.runtime_job_v2 import RuntimeJobV2
+from qiskit_ibm_runtime.runtime_options import RuntimeOptions
+from qiskit_ibm_runtime.utils.result_decoder import ResultDecoder
+
+from qiskit_serverless.core.constants import (
+    REQUESTS_TIMEOUT,
+    ENV_JOB_GATEWAY_TOKEN,
+    ENV_JOB_GATEWAY_HOST,
+    ENV_JOB_ID_GATEWAY,
+    ENV_GATEWAY_PROVIDER_VERSION,
+    GATEWAY_PROVIDER_VERSION_DEFAULT,
+)
+
+
+def associate_runtime_job_with_serverless_job(
+    runtime_job_id: str, session_id: Optional[str] = None
+) -> bool:
+    """Make a request to gateway to associate runtime job id with serverless job id.
+
+    Args:
+        runtime_job_id (str): job id for runtime primitive
+        session_id (str): session/batch id
+
+    Returns:
+        bool: if request was ok
+    """
+    version = os.environ.get(ENV_GATEWAY_PROVIDER_VERSION)
+    if version is None:
+        version = GATEWAY_PROVIDER_VERSION_DEFAULT
+
+    token = os.environ.get(ENV_JOB_GATEWAY_TOKEN)
+    if token is None:
+        logging.warning("Runtime job will not be associated with serverless job.")
+        return False
+
+    url = (
+        f"{os.environ.get(ENV_JOB_GATEWAY_HOST)}/"
+        f"api/{version}/jobs/{os.environ.get(ENV_JOB_ID_GATEWAY)}/add_runtimejob/"
+    )
+    response = requests.post(
+        url,
+        data={"runtime_job": runtime_job_id, "session_id": session_id},
```

Comment:
Is it possible to use `json` here to run a standard json call? (Maybe not)

Reply:
not needed, since requests doing it under the hood

Reply:
Correct me if I'm wrong here @IceKhan13 , in https://github.com/Qiskit/qiskit-serverless/pull/1285 we needed to change `data` by `json` because `requests` was not using `content-type` JSON by default:

https://requests.readthedocs.io/en/latest/user/quickstart/#more-complicated-post-requests

Reply:
OMG, that what you meant :) 

I though you wanted to wrap payload in `json.dumps` :) 

added in https://github.com/Qiskit/qiskit-serverless/pull/1358/commits/e96c3470bf1c431fc83b2bfbc736eebe522e09de

Reply:
well, you know, always the best one explaining itself, sorry 😅